web/src/RootSwitcher.tsx

```diff
@@ -37,11 +39,11 @@ const LicensesPage = lazyWithRetry(() => import('./routes/LicensesPage'))
 const RootSwitcher = ({ setContentLanguage }: RootSwitcherProps): ReactElement => {
   const { i18n } = useTranslation()
   const { fixedCity, cityNotCooperating, jpalTracking } = buildConfig().featureFlags
-  const languageCode = useMatch('/:slug/:languageCode/*')?.params.languageCode
+  const { routeParam0, routeParam1, '*': splat } = useMatch('/:routeParam0/:routeParam1/*')?.params ?? {}
   useScrollToTop()
 
   const detectedLanguageCode = i18n.language
-  const language = languageCode ?? detectedLanguageCode
+  const language = routeParam1 ?? detectedLanguageCode
```

Comment:
❓ Doesn't this PR make sure that `routeParam1` exactly isn't necessarily the language?

Reply:
Yes, but the case that it is a feature slug (e.g. `news`) is handled by the redirects in which case this will not make a difference. In every other case it is either
* a correct language, then everything will continue to work as expected
* not a correct language, than the `LanguageNotAvailablePage` will be displayed from which the user can choose the preferred language